src/Compilers/Core/Portable/SourceGeneration/Nodes/IIncrementalGeneratorNode.cs

```diff
@@ -2,23 +2,16 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System.Collections.Generic;
-using System.Threading;
+using Microsoft.CodeAnalysis.PooledObjects;
 
-namespace Microsoft.CodeAnalysis
-{
-    /// <summary>
-    /// Represents a node in the execution pipeline of an incremental generator
-    /// </summary>
-    /// <typeparam name="T">The type of value this step operates on</typeparam>
-    internal interface IIncrementalGeneratorNode<T>
-    {
-        NodeStateTable<T> UpdateStateTable(DriverStateTable.Builder graphState, NodeStateTable<T>? previousTable, CancellationToken cancellationToken);
-
-        IIncrementalGeneratorNode<T> WithComparer(IEqualityComparer<T> comparer);
+namespace Microsoft.CodeAnalysis;
 
-        IIncrementalGeneratorNode<T> WithTrackingName(string name);
+/// <summary>
+/// Represents a node in the execution pipeline of an incremental generator.
+/// </summary>
+internal interface IIncrementalGeneratorNode
+{
+    TransformFactory? TransformFactory { get; }
```

Comment:
📝 Another option is to create an `AbstractTransformFactory` with two derived types: `DefaultTransformFactory` and `CachingTransformFactory`. Types would use the non-caching `DefaultTransformFactory` unless assigned a caching one. This would avoid some of the `?.` and `??` usage seen throughout this PR.

Reply:
i like the idea of a sentinel 'default' transformer to make things simpler.

Reply:
➡️ Updated to make TransformFactory required for node use